File: redux_bot_js/redux_bot.js
// redux_bot.js — Bot Discord em JavaScript
// Instalar: npm install discord.js axios
// Rodar:    node redux_bot.js

const { Client, GatewayIntentBits, EmbedBuilder } = require('discord.js');
const axios = require('axios');

// ── CONFIGURACAO ──────────────────────────────
const DISCORD_TOKEN = 'SEU_TOKEN_DISCORD';
const ADMIN_TOKEN   = 'REDUX_ADMIN_2026';
const API_BASE      = 'https://reduxadminkey.netlify.app/api';
const PREFIX        = '!';
// ─────────────────────────────────────────────

const client = new Client({
  intents: [
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.MessageContent,
    GatewayIntentBits.DirectMessages
  ]
});

function embedOk(title, desc) {
  return new EmbedBuilder().setColor(0x2233CC).setTitle('✅ ' + title).setDescription(desc).setFooter({ text: 'Redux Key System' });
}
function embedErr(title, desc) {
  return new EmbedBuilder().setColor(0xCC2233).setTitle('❌ ' + title).setDescription(desc).setFooter({ text: 'Redux Key System' });
}
function embedKey(key, type, expiry) {
  const color  = type === 'premium' ? 0xDDAA33 : 0x44FF88;
  const tStr   = type === 'premium' ? '★ Premium' : '● Free';
  const eStr   = expiry === 'unlimited' ? '∞ Ilimitado' : expiry;
  return new EmbedBuilder()
    .setColor(color).setTitle('🔑 Sua Redux Key')
    .addFields(
      { name: 'Key',    value: '`' + key + '`', inline: false },
      { name: 'Tipo',   value: tStr,             inline: true },
      { name: 'Expira', value: eStr,             inline: true }
    )
    .setFooter({ text: 'Nao compartilhe sua key!' });
}

client.once('ready', () => {
  console.log('[Redux Bot] Online como ' + client.user.tag);
  client.user.setActivity('Redux Key System', { type: 2 });
});

client.on('messageCreate', async (msg) => {
  if (msg.author.bot || !msg.content.startsWith(PREFIX)) return;
  const args = msg.content.slice(PREFIX.length).trim().split(/\s+/);
  const cmd  = args.shift().toLowerCase();

  // !getkey [premium|free]
  if (cmd === 'getkey') {
    const tipo = (args[0] || 'premium').toLowerCase();
    if (tipo !== 'premium' && tipo !== 'free')
      return msg.reply({ embeds: [embedErr('Tipo invalido', 'Use `!getkey premium` ou `!getkey free`')] });
    try {
      const { data } = await axios.get(API_BASE + '/generate', {
        params: { type: tipo, user: msg.author.tag, token: ADMIN_TOKEN }
      });
      if (!data.success) return msg.reply({ embeds: [embedErr('Erro', data.error || 'Falha ao gerar key')] });
      try {
        await msg.author.send({ embeds: [embedKey(data.key, data.type, data.expiry)] });
        await msg.reply({ embeds: [embedOk('Key enviada!', msg.author + ', verifique seu DM 📬')] });
      } catch {
        await msg.reply({ embeds: [embedKey(data.key, data.type, data.expiry)] });
      }
    } catch (err) {
      msg.reply({ embeds: [embedErr('Erro de servidor', err.message)] });
    }
  }

  // !validatekey <key>
  else if (cmd === 'validatekey') {
    const key = args[0];
    if (!key) return msg.reply({ embeds: [embedErr('Uso', '`!validatekey KEYP_...`')] });
    try {
      const { data } = await axios.get(API_BASE + '/validate', { params: { key, hwid: '' } });
      if (data.valid)
        msg.reply({ embeds: [embedOk('Key valida', '**Tipo:** ' + data.type + '\n**Key:** `' + key + '`')] });
      else
        msg.reply({ embeds: [embedErr('Key invalida', '**Motivo:** ' + data.error + '\n**Key:** `' + key + '`')] });
    } catch (err) { msg.reply({ embeds: [embedErr('Erro', err.message)] }); }
  }

  // !revokekey <key>
  else if (cmd === 'revokekey') {
    const key = args[0];
    if (!key) return msg.reply({ embeds: [embedErr('Uso', '`!revokekey KEYP_...`')] });
    try {
      const { data } = await axios.get(API_BASE + '/revoke', { params: { key, token: ADMIN_TOKEN } });
      if (data.success)
        msg.reply({ embeds: [embedOk('Key revogada', 'A key `' + key + '` foi revogada.')] });
      else
        msg.reply({ embeds: [embedErr('Erro', data.error)] });
    } catch (err) { msg.reply({ embeds: [embedErr('Erro', err.message)] }); }
  }

  // !resethwid <key>
  else if (cmd === 'resethwid') {
    const key = args[0];
    if (!key) return msg.reply({ embeds: [embedErr('Uso', '`!resethwid KEYP_...`')] });
    try {
      const { data } = await axios.get(API_BASE + '/reset-hwid', { params: { key, token: ADMIN_TOKEN } });
      if (data.success)
        msg.reply({ embeds: [embedOk('HWID resetado', 'HWID da key `' + key + '` resetado.\nUsuario pode usar em outro PC.')] });
      else
        msg.reply({ embeds: [embedErr('Erro', data.error)] });
    } catch (err) { msg.reply({ embeds: [embedErr('Erro', err.message)] }); }
  }

  // !keyhelp
  else if (cmd === 'keyhelp') {
    msg.reply({ embeds: [
      new EmbedBuilder().setColor(0x2233CC).setTitle('Redux Key System — Comandos')
        .addFields(
          { name: '`!getkey premium`',   value: 'Gera key premium e envia no DM',       inline: false },
          { name: '`!getkey free`',      value: 'Gera key free e envia no DM',          inline: false },
          { name: '`!validatekey <key>`', value: 'Verifica se uma key e valida',        inline: false },
          { name: '`!revokekey <key>`',   value: 'Revoga uma key (admin)',              inline: false },
          { name: '`!resethwid <key>`',   value: 'Reseta HWID de uma key (admin)',      inline: false }
        ).setFooter({ text: 'Redux Key System' })
    ]});
  }
});

client.login(DISCORD_TOKEN);

File: redux-final(1)/redux-final/redux_bot_py/redux_bot.py
# redux_bot.py — Bot Discord em Python
# Instalar: pip install discord.py requests
# Rodar:    python redux_bot.py

import discord
import requests
from discord.ext import commands

# ── CONFIGURACAO ──────────────────────────────
DISCORD_TOKEN = 'SEU_TOKEN_DISCORD'
ADMIN_TOKEN   = 'REDUX_ADMIN_2026'
API_BASE      = 'https://reduxadminkey.netlify.app/api'
PREFIX        = '!'
# ─────────────────────────────────────────────

intents = discord.Intents.default()
intents.message_content = True
bot = commands.Bot(command_prefix=PREFIX, intents=intents)

def embed_ok(title, desc):
    e = discord.Embed(title='✅ ' + title, description=desc, color=0x2233CC)
    e.set_footer(text='Redux Key System')
    return e

def embed_err(title, desc):
    e = discord.Embed(title='❌ ' + title, description=desc, color=0xCC2233)
    e.set_footer(text='Redux Key System')
    return e

def embed_key(key, key_type, expiry):
    color = 0xDDAA33 if key_type == 'premium' else 0x44FF88
    t_str = '★ Premium' if key_type == 'premium' else '● Free'
    e_str = '∞ Ilimitado' if expiry == 'unlimited' else expiry
    e = discord.Embed(title='🔑 Sua Redux Key', color=color)
    e.add_field(name='Key',    value=f'`{key}`', inline=False)
    e.add_field(name='Tipo',   value=t_str,      inline=True)
    e.add_field(name='Expira', value=e_str,       inline=True)
    e.set_footer(text='Nao compartilhe sua key!')
    return e

@bot.event
async def on_ready():
    print(f'[Redux Bot] Online como {bot.user}')
    await bot.change_presence(activity=discord.Activity(
        type=discord.ActivityType.listening, name='Redux Key System'))

@bot.command(name='getkey')
async def getkey(ctx, tipo: str = 'premium'):
    tipo = tipo.lower()
    if tipo not in ('premium', 'free'):
        return await ctx.reply(embed=embed_err('Tipo invalido', 'Use `!getkey premium` ou `!getkey free`'))
    try:
        r    = requests.get(f'{API_BASE}/generate',
                            params={'type': tipo, 'user': str(ctx.author), 'token': ADMIN_TOKEN},
                            timeout=10)
        data = r.json()
    except Exception as ex:
        return await ctx.reply(embed=embed_err('Erro de servidor', str(ex)))

    if not data.get('success'):
        return await ctx.reply(embed=embed_err('Erro', data.get('error', 'Falha ao gerar key')))

    emb = embed_key(data['key'], data['type'], data['expiry'])
    try:
        await ctx.author.send(embed=emb)
        await ctx.reply(embed=embed_ok('Key enviada!', f'{ctx.author.mention}, verifique seu DM 📬'))
    except discord.Forbidden:
        await ctx.reply(embed=emb)

@bot.command(name='validatekey')
async def validatekey(ctx, key: str = None):
    if not key:
        return await ctx.reply(embed=embed_err('Uso', '`!validatekey KEYP_...`'))
    try:
        r    = requests.get(f'{API_BASE}/validate', params={'key': key, 'hwid': ''}, timeout=10)
        data = r.json()
    except Exception as ex:
        return await ctx.reply(embed=embed_err('Erro de servidor', str(ex)))

    if data.get('valid'):
        await ctx.reply(embed=embed_ok('Key valida', f'**Tipo:** {data["type"]}\n**Key:** `{key}`'))
    else:
        await ctx.reply(embed=embed_err('Key invalida', f'**Motivo:** {data.get("error")}\n**Key:** `{key}`'))

@bot.command(name='revokekey')
async def revokekey(ctx, key: str = None):
    if not key:
        return await ctx.reply(embed=embed_err('Uso', '`!revokekey KEYP_...`'))
    try:
        r    = requests.get(f'{API_BASE}/revoke', params={'key': key, 'token': ADMIN_TOKEN}, timeout=10)
        data = r.json()
    except Exception as ex:
        return await ctx.reply(embed=embed_err('Erro de servidor', str(ex)))

    if data.get('success'):
        await ctx.reply(embed=embed_ok('Key revogada', f'A key `{key}` foi revogada com sucesso.'))
    else:
        await ctx.reply(embed=embed_err('Erro', data.get('error', 'Falha ao revogar')))

@bot.command(name='resethwid')
async def resethwid(ctx, key: str = None):
    if not key:
        return await ctx.reply(embed=embed_err('Uso', '`!resethwid KEYP_...`'))
    try:
        r    = requests.get(f'{API_BASE}/reset-hwid', params={'key': key, 'token': ADMIN_TOKEN}, timeout=10)
        data = r.json()
    except Exception as ex:
        return await ctx.reply(embed=embed_err('Erro de servidor', str(ex)))

    if data.get('success'):
        await ctx.reply(embed=embed_ok('HWID resetado', f'HWID da key `{key}` resetado.\nUsuario pode usar em outro PC.'))
    else:
        await ctx.reply(embed=embed_err('Erro', data.get('error', 'Falha ao resetar HWID')))

@bot.command(name='keyhelp')
async def keyhelp(ctx):
    e = discord.Embed(title='Redux Key System — Comandos', color=0x2233CC)
    e.add_field(name='`!getkey premium`',    value='Gera key premium e envia no DM',     inline=False)
    e.add_field(name='`!getkey free`',       value='Gera key free e envia no DM',        inline=False)
    e.add_field(name='`!validatekey <key>`', value='Verifica se uma key e valida',       inline=False)
    e.add_field(name='`!revokekey <key>`',   value='Revoga uma key',                     inline=False)
    e.add_field(name='`!resethwid <key>`',   value='Reseta HWID de uma key',             inline=False)
    e.set_footer(text='Redux Key System')
    await ctx.reply(embed=e)

bot.run(DISCORD_TOKEN)
